Mathlib/Algebra/PUnitInstances.lean

```diff
@@ -29,65 +29,73 @@ namespace PUnit
 variable {R S : Type _} (x y : PUnit.{u + 1}) (s : Set PUnit.{u + 1})
 
 @[to_additive]
-instance : CommGroup PUnit := by
-  refine_struct
-        { mul := fun _ _ => star
-          one := star
-          inv := fun _ => star
-          div := fun _ _ => star
-          npow := fun _ _ => star
-          zpow := fun _ _ => star.. } <;>
-      intros <;>
-    exact Subsingleton.elim _ _
-
-@[simp, to_additive]
+instance commGroup: CommGroup PUnit where
+  mul := fun _ _ => PUnit.unit
+  one := PUnit.unit
+  inv := fun _ => PUnit.unit
+  div := fun _ _ => PUnit.unit
+  npow := fun _ _ => PUnit.unit
+  zpow := fun _ _ => PUnit.unit
+  mul_assoc := by intros; rfl
+  one_mul := by intros; rfl
+  mul_one := by intros; rfl
+  mul_left_inv := by intros; rfl
+  mul_comm := by intros; rfl
+
+@[to_additive (attr := simp)]
 theorem one_eq : (1 : PUnit) = star :=
   rfl
 #align punit.one_eq PUnit.one_eq
 
-@[simp, to_additive]
+@[to_additive (attr := simp)]
 theorem mul_eq : x * y = star :=
   rfl
 #align punit.mul_eq PUnit.mul_eq
 
--- `sub_eq` simplifies `punit.sub_eq`, but the latter is eligible for `dsimp`
-@[simp, nolint simp_nf, to_additive]
+-- `sub_eq` simplifies `PUnit.sub_eq`, but the latter is eligible for `dsimp`
+@[to_additive (attr := simp, nolint simpNF)]
 theorem div_eq : x / y = star :=
   rfl
 #align punit.div_eq PUnit.div_eq
 
--- `neg_eq` simplifies `punit.neg_eq`, but the latter is eligible for `dsimp`
-@[simp, nolint simp_nf, to_additive]
+-- `neg_eq` simplifies `PUnit.neg_eq`, but the latter is eligible for `dsimp`
+@[to_additive (attr := simp, nolint simpNF)]
 theorem inv_eq : x⁻¹ = star :=
   rfl
 #align punit.inv_eq PUnit.inv_eq
 
-instance : CommRing PUnit := by
-  refine' { PUnit.commGroup, PUnit.addCommGroup with natCast := fun _ => PUnit.unit.. } <;>
-      intros <;>
-    exact Subsingleton.elim _ _
-
-instance : CancelCommMonoidWithZero PUnit := by
-  refine' { PUnit.commRing with .. } <;> intros <;> exact Subsingleton.elim _ _
-
-instance : NormalizedGcdMonoid PUnit := by
-  refine' {
-          gcd := fun _ _ => star
-          lcm := fun _ _ => star
-          normUnit := fun x => 1
-          gcd_dvd_left := fun _ _ => ⟨star, Subsingleton.elim _ _⟩
-          gcd_dvd_right := fun _ _ => ⟨star, Subsingleton.elim _ _⟩
-          dvd_gcd := fun _ _ _ _ _ => ⟨star, Subsingleton.elim _ _⟩
-          gcd_mul_lcm := fun _ _ => ⟨1, Subsingleton.elim _ _⟩.. } <;>
-      intros <;>
-    exact Subsingleton.elim _ _
-
-@[simp]
+instance commRing: CommRing PUnit :=
+  { PUnit.commGroup, PUnit.addCommGroup with
```

Comment:
I see that previously, the field
```lean
natCast := fun _ => PUnit.unit
```
was manually provided in this instance. Can you include that too, just in case that decision had content?